### DISCUSSION TITLE
Need help setting up a performance file

### DISCUSSION THREAD

**Discussion by Alan-Angel** [probonopd/MiniDexed#519] [General]
Hi, So I have mini Dexed running on a pi 3, got it working with an M-Audio Oxygen controller. What I don't have is a 2*16 screen or a rotary encoder. I've also put some sysex files onto the pi. 

I can change presets but what I have no clue about is the difference between performance files and preset files in terms of selecting or changing them blindly. 

I deally I want to set up 8 bass sounds in one performance and use the sliders on the keyboard to blend them in. Can this be done? Can the keys trigger all tone generators at once and can I set up a volume slider for each one? Not sure I can have the best of both worlds because I need to use either 1 or multpiple channels.  I would do this myself but I don't know how to isolate single preset data.

Sorry this is quite a few questions in one but it will probably make sense to someone.

**Comment by probonopd** [comment]
Hi @Alan-Angel, welcome.

> I can change presets but what I have no clue about is the difference between performance files and preset files in terms of selecting or changing them blindly.

So you would like to switch performances from your MIDI Controller. Currently you can only switch voices this way, but not yet performances. You are not the first one who suggested this feature. It is being discussed here: https://github.com/probonopd/MiniDexed/discussions/496

There is an experimental build for testing in https://github.com/probonopd/MiniDexed/pull/500 - maybe you'd like to give it a try and give feedback in #500 how it is working for you?

> I can change presets but what I have no clue about is the difference between performance files and preset files in terms of selecting or changing them blindly.

There are no "presets". Only voices and performances (combinations of up to 8 voices).

> I deally I want to set up 8 bass sounds in one performance and use the sliders on the keyboard to blend them in. Can this be done? Can the keys trigger all tone generators at once and can I set up a volume slider for each one?

Yes, if you have a MIDI controller with 8 sliders that can all be set to different MIDI channels. Make a performance that contains a different bass voice for each Tone Generator (1-8) and that uses a different MIDI channel for each Tone Generator. Now, if you have a MIDI controller with 8 sliders, you need to set each slider to control the volume on a different MIDI channel.

> I would do this myself but I don't know how to isolate single preset data.

There is a PC software made by @BobanSpasic with which you can edit performances: **MiniDexedLibrarian**. For details, see https://github.com/probonopd/MiniDexed/discussions/196

**Reply by Alan-Angel** [reply]
Thanks for the detailed response. So if each tone generator is set to a different channel and I assign different volume sliders to them, can they still read the note info in some kind of Omni way? I'm pretty sure the controller will only transmit notes on one channel. I'll have to check.

**Reply by probonopd** [reply]
Each TG will only listen to the MIDI channel it is set to, or (if it is set to Omni, will listen to all channels).

Via MIDI, I don't think you can make it so that a TG reacts to all MIDI channels for some but not all MIDI messages.

**Comment by Banana71** [comment]
Hi Alan, 

> Sorry this is quite a few questions in one but it will probably make sense to someone

I'm curious and maybe there is another solution: why do you want to play 8 bass sounds on one keyboard and mix them steplessly?  What exactly is your goal with such a setup?

By the way without a display and encoder, the miniDexed can only load the "Default Performance" (sd-card root performance.ini) 
You can copy any other performance from the Performance folder to root and rename it to performance.ini and start the Raspberry.

Peter

**Reply by Alan-Angel** [reply]
Hi Peter, basically the DX bass sounds are greatly enhanced when blending together. One sound for attack, another for sustain, one an octave higher or lower etc. Having 8 sounds on sliders seems like a good way to go for live performance. I've done something similar with Synprezfm although that only layers two voices.
